src/unix/thread.c

```diff
@@ -63,7 +69,27 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
   ctx->entry = entry;
   ctx->arg = arg;
 
-  err = pthread_create(tid, NULL, uv__thread_start, ctx);
+  /* On OSX threads other than the main thread are created with a reduced stack size by default,
+   * adjust it to RLIMIT_STACK.
+   */
+#if defined(__APPLE__)
+  if (getrlimit(RLIMIT_STACK, &lim))
+    abort();
+
+  attr = &attr_storage;
+  if (pthread_attr_init(attr))
+    abort();
+
+  if (pthread_attr_setstacksize(attr, lim.rlim_cur))
```

Comment:
Do we need to `abort()` on every failure? In that case I'd just go ahead with the defaults, maybe.